src/analysis/sensitivity.jl

```diff
@@ -32,6 +32,52 @@ function ks_statistic(ks::ApproximateKSTest)::Float64
     return sqrt(n) * ks.δ
 end
 
+"""
+    _get_factor_spec(model_spec::DataFrame, factors::Vector{String})
+
+Get model spec for specified factors.
+
+# Arguments 
+- model_spec : Model specification, from `model_spec(domain)` or `rs.model_spec`
+- factors : Parameters considered for sensitivity analysis
+"""
+function _get_factor_spec(model_spec::DataFrame, factors::Vector{Symbol})
+    factors_to_assess = model_spec.fieldname .∈ [factors]
+    foi_attributes = Symbol[:fieldname, :ptype, :lower_bound, :upper_bound]
+    return model_spec[factors_to_assess, foi_attributes]
+end
+
+"""
+    _category_bins(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+
+Get number of binnings for categorical variables.
+
+# Arguments
+- S : Number of bins
+- foi_spec : Model specification for factors of interest
+- foi_cat : Vector of length `foi_cat.fieldname` which contains true where the factor is categorical and false otherwise
+"""
+function _category_bins(S::Int64, foi_spec::DataFrame, foi_cat::BitVector)
+    max_bounds = maximum(foi_spec[foi_cat, :upper_bound] .- foi_spec[foi_cat, :lower_bound])
+    return round(Int64, max(S, max_bounds))
+end
+
+"""
+    _get_cat_quantile(foi_spec::DataFrame, fact_cat::String, steps::Vector{Float64})
+
+Get quantile for categorical variable, fact_cat.
+
+# Arguments
+- foi_spec : Model specification for factors of interest
+- foi_cat : Vector of length `foi_cat.fieldname` which contains true where the factor is categorical and false otherwise
+- steps : Number of steps for defining binnings
+"""
+function _get_cat_quantile(foi_spec::DataFrame, fact_cat::Symbol, steps::Vector{Float64})
+    fact_idx = foi_spec.fieldname .== fact_cat
+    lb = foi_spec.lower_bound[fact_idx][1]
+    ub = foi_spec.upper_bound[fact_idx][1]
+    return round.(quantile(lb:ub, steps)) .- 1
+end
```

Comment:
Generally the same comment as above, but I don't understand this:

If `fact_cat` (which hasn't been updated in the argument list) is a vector of true/false, then how does the below work?

```julia
fact_idx = foi_spec.fieldname .== fact_cat
```

EDIT:
Okay, I see, `fact_cat` is the target factor name. In which case the variable name doesn't make sense to me.

Reply:
It's for factor categoical, but I'll think of another name